unleash-yggdrasil/src/lib.rs

```diff
@@ -27,13 +28,15 @@ use unleash_types::client_metrics::{MetricBucket, ToggleStats};
 
 pub type CompiledState = HashMap<String, CompiledToggle>;
 
-pub const SUPPORTED_SPEC_VERSION: &str = "4.5.2";
+pub const SUPPORTED_SPEC_VERSION: &str = "5.0.2";
+const VARIANT_NORMALIZATION_SEED: u32 = 86028157;
+
 
 pub struct CompiledToggle {
     pub name: String,
     pub enabled: bool,
     pub compiled_strategy: RuleFragment,
-    pub compiled_variant_strategy: Option<Vec<(RuleFragment, Vec<CompiledVariant>)>>,
+    pub compiled_variant_strategy: Option<Vec<(RuleFragment, Vec<CompiledVariant>, String)>>,
```

Comment:
When we are compiling variant strategy, we also need to pull out the groupId from strategy, not only variants.